### ISSUE TITLE
augmentation order

### ISSUE THREAD

**Issue by hongsukchoi** [dahiyaaneesh/peclr#4]
Hi,

According to the comments you wrote [here](https://github.com/dahiyaaneesh/peclr/blob/5992b24a6025876b3266fa6c98d0118ce60564cf/src/data_loader/sample_augmenter.py#L47), it seems like there has to be a specific order for different augmentations.

Is there any reason for this?
If I change the order, is there bad effects on the accuracy?

**Comment by dahiyaaneesh** [comment]
Yes,

This was done to ensure that artifacts in the transformed images are minimized. For example, if one crops the image first and then rotates the image by 30 degree with no scaling the image has artifacts on new corners. However rotating first followed by cropping can minimize them, as chances are high the cropping will remove the artifacts around the corner. 

To answer your second question: No we did not evaluate the effects of different ordering on accuracy but I encourage you to try that. We did qualitatively evaluate the input samples with the order mentioned in the code. This yielded good input samples with negligible artifacts that could have led to poorer performance.

I hope it helps.
